app/Jobs/ProcessPostmarkInboundWebhook.php

```diff
@@ -0,0 +1,167 @@
+     *
+     * @var string
+     */
+    public $queue = 'email';
+
+    /**
+     * The number of times the job may be attempted.
+     *
+     * @var int
+     */
+    public $tries = 1;
+
+    /**
+     * Execute the job.
+     *
+     * @phan-suppress PhanTypeArraySuspiciousNullable
+     * @phan-suppress PhanPossiblyFalseTypeArgument
+     */
+    public function handle(): void
+    {
+        $payload = $this->webhookCall->payload;
+        $subject = $payload['Subject'];
+
+        if (Str::startsWith($subject, 'Completed: ')) {
+            $summary = collect($payload['Attachments'])->firstOrFail(static function (array $value, int $key): bool {
+                return 'Summary.pdf' === $value['Name'];
+            });
+
+            $decoded = base64_decode($summary['Content'], true);
+
+            $parser = new Parser();
+
+            $pdf = $parser->parseContent($decoded);
+
+            $text = $pdf->getText();
+
+            $user = User::where(
+                'uid',
+                '=',
+                self::getValueWithRegex(
+                    '/Signed by link sent to (?P<uid>[a-z]+[0-9]+)@gatech\.edu/',
+                    $text,
+                    'uid'
+                )
+            )->sole();
+
+            $envelope = $user->envelopes()->where('complete', false)->sole();
+
+            $envelope->sent_at = self::getValueWithRegex(
+                self::SIGNER_INFO_REGEX,
+                $text,
+                'sentAt'
+            );
+
+            $envelope->viewed_at = self::getValueWithRegex(
+                self::SIGNER_INFO_REGEX,
+                $text,
+                'viewedAt'
+            );
+
+            $envelope->signed_at = self::getValueWithRegex(
+                self::SIGNER_INFO_REGEX,
+                $text,
+                'signedAt'
+            );
+
+            $envelope->completed_at = self::getValueWithRegex(
+                '/Completed\s+Security Checked\s+(?P<completedAt>\d{1,2}\/\d{1,2}\/\d{4} \d{1,2}:\d{1,2}:\d{1,2} (AM|PM))/',
+                $text,
+                'completedAt'
+            );
+
+            $envelope->signer_ip_address = self::getValueWithRegex(
+                self::SIGNER_INFO_REGEX,
+                $text,
+                'ipAddress'
+            );
+
+            $envelope->envelope_id = self::getValueWithRegex(
+                '/Envelope Id: (?P<envelopeId>[A-Z0-9]{32})/',
+                $text,
+                'envelopeId'
+            );
+
+            $envelope->url = Str::of(self::getValueWithRegex(
+                '/(?P<url>https:\/\/na3.docusign.net\/Member\/EmailStart.aspx.+)/',
+                $payload['TextBody'],
+                'url',
+                'email text'
+            ))->trim();
+
+            Storage::makeDirectory('docusign/'.$envelope->envelope_id);
+
+            collect($payload['Attachments'])->each(static function (array $value, int $key) use ($envelope): void {
+                $original_filename = $value['Name'];
```

Comment:
Is `$value['Name']` guaranteed to exist and be safe to use in a path on disk (e.g., no spaces / not vulnerable to filename attacks)?  I couldn't quite find an answer in the docs

Reply:
It is (at least theoretically) input by someone with access to configure DocuSign powerforms within apiary, so i think the attack surface is low.

Reply:
I added a regex just to be safe.